src/app/BUILD.gn

```diff
@@ -59,6 +59,7 @@ buildconfig_header("app_buildconfig") {
     "CHIP_CONFIG_ENABLE_EVENTLIST_ATTRIBUTE=${enable_eventlist_attribute}",
     "CHIP_CONFIG_ENABLE_ICD_SERVER=${chip_enable_icd_server}",
     "CHIP_CONFIG_ENABLE_READ_CLIENT=${chip_enable_read_client}",
+    "ICD_REPORT_ON_ENTER_ACTIVE_MODE=${chip_report_on_active_mode}",
```

Comment:
Do we use `ifdef` to check the define or `if`?

Reply:
`#if ICD_REPORT_ON_ENTER_ACTIVE_MODE`